test/server/cards/15.1-TL/ProceduralInterferece.spec.js

```diff
@@ -0,0 +1,89 @@
+            this.shameful2 = this.player1.findCardByName('shameful-display', 'province 1');
+            this.silent = this.player2.findCardByName('silent-ones-monastery', 'province 2');
+
+            this.player2.reduceDeckToNumber('dynasty deck', 0);
+            this.player2.moveCard(this.aranat2, 'province 1');
+            this.player2.moveCard(this.ballista2, 'dynasty deck');
+
+            this.player1.reduceDeckToNumber('dynasty deck', 0);
+            this.player1.moveCard(this.interference, 'province 1');
+            this.player1.moveCard(this.interference2, 'province 1');
+            this.player1.moveCard(this.uji, 'dynasty deck');
+        });
+
+        it('should prompt a province to be chosen', function () {
+            this.player1.clickCard(this.interference);
+            expect(this.player1).toBeAbleToSelect(this.shamefulDisplay);
+        });
+
+        it('should give the opponent an option to select between effects', function() {
+            this.player1.clickCard(this.interference);
+            this.player1.clickCard(this.shamefulDisplay);
+
+            expect(this.player2).toHavePromptButton('discard each card in the province');
+            expect(this.player2).toHavePromptButton('let opponent gain 2 honor');
+        });
+
+        it('should discard all cards in the province when selected', function() {
+            this.player1.clickCard(this.interference);
+            this.player1.clickCard(this.shamefulDisplay);
+            this.player2.clickPrompt('Discard each card in the province');
+
+            expect(this.aranat2.location).toBe('dynasty discard pile');
+            expect(this.ballista2.location).toBe('province 1');
+            expect(this.ballista2.facedown).toBe(true);
+            expect(this.getChatLogs(5)).toContain('player1 plays Procedural Interference to discard each card in the province Shameful Display');
+        });
+
+        it('should let the player gain 2 honor', function() {
+            this.player1.clickCard(this.interference);
+            this.player1.clickCard(this.shamefulDisplay);
+            this.player2.clickPrompt('let opponent gain 2 honor');
+
+            expect(this.player1.player.honor).toBe(12);
+            expect(this.player2.player.honor).toBe(10);
+        });
+
+        it('should not let you gain 4 honor when Silent Ones is present', function() {
+            this.player1.clickCard(this.interference);
+            this.player1.clickCard(this.shamefulDisplay);
+            this.player2.clickPrompt('let opponent gain 2 honor');
+            this.player1.clickCard(this.interference2);
+            expect(this.player2).not.toHavePromptButton('let opponent gain 2 honor');
+        });
+
+        it('should not let you choose your own province', function() {
+            this.player1.clickCard(this.interference);
+            expect(this.player1).not.toBeAbleToSelect(this.shameful2);
+        });
+
```

Comment:
Can you add a test that the Stronghold province can't be chosen? (a province with no cards to discard should be an illegal target)